### PR TITLE
Wire label and call cache diff endpoint. Closes #2431 and #2430

### PR DESCRIPTION
**Pull request by geoffjentry**

### CODE REVIEW

File: src/main/resources/swagger/cromiam.yaml [hunk 1]
```diff
@@ -823,3 +916,14 @@ definitions:
...
+    required:
+      - id
+      - labels
+    properties:
+      id:
+        type: string
+        description: The identifier of the workflow
+      labels:
+        type: string
+        description: The custom labels as json
```
Comment (mcovarr) [inline src/main/resources/swagger/cromiam.yaml]: \n

File: src/main/scala/cromiam/webservice/CromIamApiService.scala [hunk 2]
```diff
@@ -54,19 +55,21 @@ trait CromIamApiService extends Directives with SprayJsonSupport with DefaultJso
     Http().singleRequest(cromwellRequest)
   }
 
-  private[webservice] def authorizeThenForwardToCromwell(user: Option[String], workflowId: String, action: String, request: HttpRequest): Future[HttpResponse] = {
+  private[webservice] def authorizeThenForwardToCromwell(user: Option[String], workflowIds: List[String], action: String, request: HttpRequest): Future[HttpResponse] = {
+    val authCheck = Future.sequence(workflowIds map { w => requestAuth(WorkflowAuthorizationRequest(user.getOrElse("anon"), w, action)) })
+
     (for {
-      _ <- requestAuth(WorkflowAuthorizationRequest(user.getOrElse("anon"), workflowId, action))
+      _ <- authCheck
```
Comment (mcovarr) [inline src/main/scala/cromiam/webservice/CromIamApiService.scala]:
```
import cats.instances.future._
import cats.instances.list._
import cats.syntax.traverse._
...
    def authForId(id: String) = requestAuth(WorkflowAuthorizationRequest(user.getOrElse("anon"), id, action))

    (for {
      _ <- workflowIds traverse authForId

...
```

File: src/main/scala/cromiam/webservice/CromIamApiService.scala [hunk 3]
```diff
@@ -113,9 +116,12 @@ trait CromIamApiService extends Directives with SprayJsonSupport with DefaultJso
     directive { handleRequest(f) }
   }
 
-  def workflowGetRoute(urlSuffix: String): Route = path("api" / "workflows" / Segment / Segment / urlSuffix) { (_, workflowId) =>
-    handleRequest(get) { (userId, req) => authorizeReadThenForwardToCromwell(userId, workflowId, req) }
+  def workflowGetRoute(urlSuffix: String): Route = workflowRoute(urlSuffix, get)
```
Comment (mcovarr) [inline src/main/scala/cromiam/webservice/CromIamApiService.scala:123]: nice DRYing!

**Review by mcovarr — APPROVED** [review]

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/CromIAM/pull-request/13/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/CromIAM)

**Review by kshakir — APPROVED** [review]

**Comment by kshakir** [comment]
👍

[Approved with PullApprove](https://pullapprove.com/broadinstitute/CromIAM/pull-request/13/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/CromIAM)

**Pull request merged**